osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -181,8 +181,9 @@ protected override void HandleItemActivated(CarouselItem item)
                     return;
 
                 case BeatmapSetInfo setInfo:
-                    // Selecting a set isn't valid – let's re-select the first difficulty.
-                    CurrentSelection = setInfo.Beatmaps.First();
+                    // Selecting a set isn't valid – let's re-select the first visible difficulty.
+                    if (grouping.SetItems.TryGetValue(setInfo, out var items))
+                        CurrentSelection = items.ElementAt(1).Model;
```

Comment:
This feels quite haphazard. I'd much rather this be something like

`items.Select(i => i.Model).OfType<BeatmapInfo>.First();`